prusti-encoder/src/encoders/local_def.rs

```diff
@@ -72,9 +72,10 @@ impl TaskEncoder for MirLocalDefEnc {
                 ty: vcx.alloc(ty.generic_predicate),
             }
         }
-
+        let trusted = utils::is_function_trusted(def_id);
         vir::with_vcx(|vcx| {
-            let data = if let Some(local_def_id) = def_id.as_local() {
+            let local_def_id = def_id.as_local().filter(|_| !trusted);
+            let data = if let Some(local_def_id) = local_def_id {
```

Comment:
Nit: I think we already do this somewhere else, but I don't really like this pattern of (mis)using `filter` to set an `Option` to `None` if a separate Boolean is `false`. Since we are on nightly we could just enable the [`let_chains` feature](https://dev-doc.rust-lang.org/nightly/unstable-book/language-features/let-chains.html) and write the much cleaner:
```suggestion
            let data = if !trusted && let Some(local_def_id) = def_id.as_local() {
```

Reply:
Did the change and added `#![feature(let_chains)] ` to `lib.rs`